itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/queryhistory/schema/IcebergQueryHistoryRecord.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.hadoop.hive.ql.queryhistory.schema;
+
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.hadoop.hive.ql.queryhistory.schema.QueryHistoryRecord;
+import org.apache.hadoop.hive.ql.queryhistory.schema.QueryHistorySchema;
+
+/**
+ * Convenient class for creating a QueryHistoryRecord from an iceberg Container.
+ */
+public class IcebergQueryHistoryRecord extends QueryHistoryRecord {
```

Comment:
why do we need new DTO if it's just a simple builder? 
````
new QueryHistoryRecordBuilder(icebergRecord).build()
````

Reply:
this specific to single test class (TestIcebergRepository), so if this is overengineering, then QueryHistoryRecordBuilder is also overengineering, let me refactor this to a simple method in the test class

Reply:
hm, wait, for this method I need to a) open the method:
```
  private void set(Field field, Object value) {
    record.put(field.name, value);
  }
```
or b) define QueryHistoryRecordBuilder in the Record class, which is not okay, it cannot contain iceberg-specific stuff

what do you suggest here?

Reply:
let's leave it since it's a test stuff